vignettes/pkgnet-intro.Rmd

```diff
@@ -0,0 +1,249 @@
+
+As a sample, here's how the **Function Network Visualization** looks for `pkgnet`: 
+
+<!--  SIDE BY SIDE START  -->
+<!--  Take care to ensure equal lengths  -->
+<div class="column-left">
+#### Default
+```{r demoVis1, echo=FALSE}
+reportObj$PackageFunctionReporter$graph_viz
+```
+Functional dependencies are visible.  For example, we can see that both `batting_avg` and `slugging_avg` functions depend upon the `at_bats` function.  
+
+We also see that nothing depends on the `on_base_pct` function.  This might be valuable information to an R package developer. 
+</div>
+
+<div class="column-right">
+#### With Coverage Information 
+```{r demoVis2, echo=FALSE}
+reportObjWithCovr$PackageFunctionReporter$graph_viz
+```
+Same as the default visualization except we can see coverage information as well (Red = 0%, Green = 100%). 
+
+It appears the function that most other functions depend upon, `at_bats`, is well covered.  However, no other functions are covered by unit tests.
+</div>
+
+<!--  SIDE BY SIDE END -->
+
+<br><br>
+
+#### Check out the full HTML report for more results  
+
+## The List Object
+
+The `CreatePackageReport()` function returns a list with three items: 
+
+1. PackageSummaryReporter  
+2. PackageDependencyReporter  
+3. PackageFunctionReporter  
+
+Each items contains information visible in the report *and more*.  We can use this information for a more detailed analysis of the results and/or more easily incorporate `pkgnet` results into other R processes.
+
+Here are a few notable items available within the list object.
+
+### Node Information
+Both the `PackageDependencyReporter` and the `PackageFunctionReporter` contain metrics about their package dependencies or functions (a.k.a network nodes) in a `nodes` table.  
+
+```{r nodes}
+dim(reportObjWithCovr$PackageFunctionReporter$nodes)
+names(reportObjWithCovr$PackageFunctionReporter$nodes)
+```
+
+### Network Measures 
+Both the `PackageDependencyReporter` and the `PackageFunctionReporter` contain measure based on their network structure in a `network_measures` list. 
+
+```{r networkMeasures}
+reportObjWithCovr$PackageFunctionReporter$network_measures
+```
+
+### Network Object
+Both the `PackageDependencyReporter` and the `PackageFunctionReporter` are available as [igraph](http://igraph.org/r/) objects named `pkg_graph`
+
+```{r networkObj}
+reportObjWithCovr$PackageFunctionReporter$pkg_graph
+```
+
+***
+# A Deeper Look
+With the reports and objects produced by `pkgnet` by default, there is plenty to inform us on the inner workings of an R package.  However, we may want to know MORE! Since the [igraph](http://igraph.org/r/) objects are available, we can leverage those for further analysis.  Here are some examples.  
+
+## Coverage of Most Referenced Functions
+Let's visualize the relationship between centrality and coverage in a large package such as `lubridate`.
+
+<!--  Faked Since lubridate not guaranteed to be installed at time of report creation -->
```

Comment:
proposition: what if we add one more line above this to say "if you want to run this code yourself, do the following":

```
# Create a temporary workspace
mkdir -p ~/pkgnet_example && cd ~/pkgnet_example

# Grab the lubridate source code
git clone https://github.com/tidyverse/lubridate

# Install it
cd lubridate
Rscript -e "devtools::install()"
```

and then change the path below in the code to that file path I proposed.

Thoughts? I think a vignette is more powerful if you can run the code yourself and mess with it. Like even though you're eval=FALSEing, people may still want to run the code themselves

Reply:
also I have another motivation...R functions that do stuff to packages living on the local file system are really inconsistent about:
1) whether they expect a package name or a file path
2) whether that file path should be to the folder _containing_ the package or to the root of the package

by having a walkthrough people can actually run if they want, we can help eliminate some of that ambiguity